modules/building-operator-with-helm-using-osdk.adoc

```diff
@@ -0,0 +1,352 @@
+resources cluster-wide. Namespace-scoped operators are preferred because of
+their flexibility. They enable decoupled upgrades, namespace isolation for
+failures and monitoring, and differing API definitions.
++
+However, there are use cases where a cluster-scoped operator may make sense. For
+example, the `cert-manager` operator is often deployed with cluster-scoped
+permissions and watches so that it can manage issuing certificates for an entire
+cluster.
++
+If you would like to create your `nginx-operator` project to be cluster-scoped,
+use the following `operator-sdk new` command instead:
++
+----
+$ operator-sdk new nginx-operator \
+    --cluster-scoped --api-version=example.com/v1alpha1 \
+    --kind=Nginx --type=helm
+----
++
+Using the `--cluster-scoped` flag will scaffold the new Operator with the
+following modifications:
++
+--
+* `deploy/operator.yaml`: Set `WATCH_NAMESPACE=""` instead of setting it to the
+Pod's namespace.
+* `deploy/role.yaml`: Use `ClusterRole` instead of `Role`.
+* `deploy/role_binding.yaml`:
+** Use `ClusterRoleBinding` instead of `RoleBinding`.
+** Set the subject namespace to `REPLACE_NAMESPACE`. This must be changed to the
+namespace in which the Operator is deployed.
+--
+
+. *Customize the Operator logic.*
++
+For this example, the `nginx-operator` executes the following reconciliation
+logic for each Nginx Custom Resource (CR):
++
+--
+* Create a Nginx Deployment if it does not exist.
+* Create a Nginx Service if it does not exist.
+* Create a Nginx Ingress if it is enabled and does not exist.
+* Ensure that the Deployment, Service, and optional Ingress match the desired
+configuration (e.g., replica count, image, service type, etc.) as specified by
+the Nginx CR.
+--
++
+By default, the `nginx-operator` watches `Nginx` resource events as shown in the
+`watches.yaml` file and executes Helm releases using the specified chart:
++
+----
+---
+- version: v1alpha1
+  group: example.com
+  kind: Nginx
+  chart: /opt/helm/helm-charts/nginx
+----
+
+.. *Review the Nginx Helm chart.*
++
+When a Helm Operator project is created, the Operator SDK creates an example Helm chart that contains a set of templates for a simple Nginx release.
++
+For this example, templates are available for Deployment, Service, and Ingress
+resources, along with a `NOTES.txt` template, which Helm chart developers use to
+convey helpful information about a release.
++
+If you are not already familiar with Helm Charts, take a moment to review the
+link:https://docs.helm.sh/developing_charts/[Helm Chart developer documentation].
+
+.. *Understand the Nginx CR spec.*
++
+Helm uses a concept called
+link:https://docs.helm.sh/using_helm/#customizing-the-chart-before-installing[values]
+to provide customizations to a Helm chart's defaults, which are defined in the
+Helm chart's `values.yaml` file.
++
+Overriding these defaults is as simple as setting the desired values in the CR
```

Comment:
Override these defaults by setting the desired values . . .

(It's good practice to avoid labelling processes as simple because if the user doesn't find the process simple, then they can feel silly or frustrated.)